src/game/objects/har.c

```diff
@@ -797,32 +828,32 @@ void har_take_damage(object *obj, const str *string, float damage, float stun) {
     // interrupted
     h->executing_move = 0;
 
-    if(h->linked_obj) {
-        object *linked = game_state_find_object(obj->gs, h->linked_obj);
-        if(linked) {
-            // end the animation of the linked object, so it can go to the successor
-            linked->animation_state.finished = 1;
-        }
-    }
+    game_player *other_player = game_state_get_player(obj->gs, !h->player_id);
+    object *other_har = game_state_find_object(obj->gs, other_player->har_obj_id);
 
-    if(vector_size(&h->child_objects)) {
-        // shadow children need to die when the controlling HAR is hit
+    vector vec;
+    vector_create(&vec, sizeof(object *));
+    if(har_find_linked_objects(obj, &vec)) {
+        log_warn("found linked objects");
         iterator it;
-        uint32_t *child_id;
-        vector_iter_begin(&h->child_objects, &it);
-        foreach(it, child_id) {
-            object *child = game_state_find_object(obj->gs, *child_id);
-            if(child) {
-                int next_anim = af_get_move(h->af_data, child->cur_animation->id)->throw_duration;
-                if(next_anim) {
-                    object_set_animation(child, &af_get_move(h->af_data, next_anim)->ani);
-                } else {
-                    child->animation_state.finished = 1;
-                }
+        vector_iter_begin(&vec, &it);
+        object **linked;
+        foreach(it, linked) {
+            log_warn("linked object has flags %d", (*linked)->object_flags);
+            // if MC and UZ is set, switch the object's gravity to the owning HAR's gravity
+            if((*linked)->object_flags & OBJECT_FLAGS_MC && (*linked)->object_flags & OBJECT_FLAGS_UZ) {
```

Comment:
```suggestion
            if((*linked)->object_flags & OBJECT_FLAGS_MC && (*linked)->object_flags & OBJECT_FLAGS_UD) {
```

This got mixed up again in the rebase